src/pubsub.js

```diff
@@ -203,6 +209,46 @@ class BasicPubSub extends Pubsub {
     this._handleRpcControl(peer, rpc)
   }
 
+  /**
+   * Validates the given message.
+   * @param {Peer} peer
+   * @param {rpc.RPC.Message} message
+   * @returns {Promise<Boolean>}
+   */
+  async validate (peer, message) {
+    const isValid = await super.validate(message)
+    if (!isValid) {
+      return false
+    }
+    for (const topic of message.topicIDs) {
+      const validatorFn = this.topicValidators.get(topic)
+      if (validatorFn) {
+        const result = validatorFn(topic, peer, message)
```

Comment:
it's more likely to me to be a transformer function instead of a validator because the result is not only true/false

Reply:
Yeah, esp in gossipsub v1.1, with "extended validators", results can be more than just true/false.
I'm planning on overriding `_processTopicValidatorResult` to handle these "extended validators". Other protocols might have different handling, so they can ofc override with protocol-specific handling.

The reason I named it 'validator' rather than transformer is to align with the gossipsub v1.0 and v1.1 spec. (see here: https://github.com/libp2p/specs/tree/master/pubsub#topic-validation and here: https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.1.md#extended-validators)

Reply:
thanks, that's good to me